### PR TITLE
Minor node clean-ups

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt [hunk 1]
```diff
@@ -742,7 +736,7 @@ private fun PackageJson.getScopeDependencies(type: YarnDependencyType) =
 private fun getYarnExecutable(workingDir: File): File {
     val yarnrcFile = workingDir.resolve(YARN2_RESOURCE_FILE)
     val yarnConfig = Yaml.default.parseToYamlNode(yarnrcFile.readText()).yamlMap
-    val yarnPath = yarnConfig.get<YamlScalar>(YARN_PATH_PROPERTY_NAME)?.content
+    val yarnPath = yarnConfig.get<YamlScalar>("yarnPath")?.content
```
Comment (mnonnenmacher) [inline plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:680]: Commit message nit: Spell out "WRT".

File: plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt [hunk 2]
```diff
@@ -205,6 +177,13 @@ class Yarn2(
 
     override fun getVersionRequirement(): RangesList = RangesListFactory.create(">=2.0.0")
 
+    private fun isCorepackEnabled(workingDir: File): Boolean =
+        if (OPTION_COREPACK_OVERRIDE in options) {
+            options[OPTION_COREPACK_OVERRIDE].toBoolean()
+        } else {
+            isCorepackEnabledInManifest(workingDir)
+        }
```
Comment (mnonnenmacher) [inline plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:186]: The extraction of this function is unrelated to the commit message.

File: plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt
Comment (mnonnenmacher) [inline plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:1]: Commit message: Typo in "distinction"

File: plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt [hunk 3]
```diff
@@ -653,23 +653,21 @@ private data class AdditionalData(
...
- * Clean the [rawVersion] string contained in a Yarn 2+ locator to have it compatible with NPM/Semver.
+ * Clean this Yarn2 version string (originating from a Yarn 2+ locator) for compatibility with NPM/Semver.
  */
-private fun cleanYarn2VersionString(rawVersion: String): String {
-    // 'Patch' locators are complex expressions such as
-    // resolve@npm%3A2.0.0-next.3#~builtin<compat/resolve>%3A%3Aversion=2.0.0-next.3&hash=07638b
-    // Therefore, the version has to be extracted (here '2.0.0-next.3').
-    var result = rawVersion.substringAfter("version=")
+private fun String.cleanVersionString(): String =
+    this
```
Comment (mnonnenmacher) [inline plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:658]: `this` is redundant, are you keeping it for formatting reasons?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: plugins/package-managers/node/src/main/kotlin/Npm.kt [hunk 4]
```diff
@@ -199,7 +199,7 @@ open class Npm(
         // Create packages for all modules found in the workspace and add them to the graph builder. They are
         // reused when they are referenced by scope dependencies.
         val packages = parseInstalledModules(workingDir)
-        graphBuilder.addPackages(packages.values)
+        graphBuilder.addPackages(packages)
```
Comment (sschuberth) [inline plugins/package-managers/node/src/main/kotlin/Npm.kt:202]: Commit message: Typo in "repspectively"

File: plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt
Comment (sschuberth) [reply plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:1]: Another nit: You could explain in what regard consistency is improved, i.e. that usually we don't have constants for thins that are only used in a single place.

However, as you're removing a `public` constant here, this is strictly speaking a breaking change and should have `!` in the title.

File: plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt [hunk 5]
```diff
@@ -71,6 +70,26 @@ import org.ossreviewtoolkit.utils.ort.showStackTrace
 import org.semver4j.RangesList
 import org.semver4j.RangesListFactory
 
+/**
+ * The name of Yarn 2+ resource file.
+ */
+private const val YARN2_RESOURCE_FILE = ".yarnrc.yml"
```
Comment (sschuberth) [inline plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:77]: Also here, making this `private` is strictly speaking a breaking change.

**Comment by sschuberth** [comment]
As this PR introduces changes that could be marked as breaking, I propose to postpone the merge of it until after today's fixup release of ORT to not have a major version bump.

File: plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt (hunk 3 above)
Comment (fviernau) [reply plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:658]: yes.

Comment (mnonnenmacher) [reply plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt:658]: Ok, I guess I'm fine with that.

**Review by mnonnenmacher — APPROVED** [review]
